debezium-core/src/main/java/io/debezium/data/geometry/Geometry.java

```diff
@@ -33,9 +33,9 @@ public class Geometry {
      *
      * @return the schema builder
      */
-    public static SchemaBuilder builder() {
+    public static SchemaBuilder builder(String columnName) {
         return SchemaBuilder.struct()
-                .name(LOGICAL_NAME)
+                .name(LOGICAL_NAME + "__" + columnName) // temporary fix for DBZ-6964
```

Comment:
This is the essence of the change. I don't feel super happy about it and looking at the Cassandra PR https://github.com/debezium/debezium-connector-cassandra/pull/121 there might be other fields affected in general.
Looking at the registry code it seems that all types that can collide, collide when there are other parameters used than things like precision and scale. For example DECIMAL types are handled here:
https://github.com/Apicurio/apicurio-registry/blob/f2d1f06cae0b709acf3f6d4edb982f5775b50fa9/utils/converter/src/main/java/io/apicurio/registry/utils/converter/avro/AvroData.java#L886-L896

Reply:
Making logical names dynamic will break the JDBC sink connector.   If we go with this approach, we'll need to add a check for "prefixed geometry logical names" to sanitize the incoming data to correctly lookup and resolve target columns accordingly.

Reply:
@Naros I agree. But let me challenge this. Is this the right way to to it? Couldn't we use `"__debezium.source.column.type" : "GEOMETRY",` schema parameter instead? With this change we can guarantee that all parameters are stored in the registry. Then we can rely on our custom fields to decide on the Debezium datatype instead of relying on the default behavior of the registry.

Reply:
We could for Debezium events, but don't forget the JDBC sink can be used by non-Debezium producers.

We will need to look at the schema type no matter what for the latter use case; I just want to be mindful of any changes we make here and how that could impact performance, specifically given how hard Mario worked on improving the sink's performance metrics already; no need to walk that back.

Reply:
@Naros  `and resolve target columns accordingly`

I think the jdbc sink resolves the column name from `field.name()` not from the schema, does it? The issue will be with identifying the correct data type:

Looking at the code in `iGeneralDatabaseDialect#getSchemaType()`

```
      if (!Objects.isNull(schema.name())) {
            final Type type = typeRegistry.get(schema.name());
            if (!Objects.isNull(type)) {
                LOGGER.trace("Schema '{}' resolved by name from registry to type '{}'", schema.name(), type);
                return type;
            }
        }
        if (!Objects.isNull(schema.parameters())) {
            final String columnType = schema.parameters().get("__debezium.source.column.type");
            if (!Objects.isNull(columnType)) {
                final Type type = typeRegistry.get(columnType);
```
The schema is loaded via the record's `schema.name()` (which will work because the schema comes from `field.schema()`.
But in that first case `typeRegistry.get(schema.name());` we would have to match only the first name or cleanup the key in the cache. I don't know, it is an unfortunate fix.